test/analysis/typeComplexityAnalysis.test.ts

```diff
@@ -890,6 +854,170 @@ describe('Test getQueryTypeComplexity function', () => {
             expect(nonNullMockWeightFunction.mock.calls[0].length).toBe(3);
         });
 
+        // TODO: refine complexity analysis to consider directives includes and skip
+        xdescribe('with directives @includes and @skip', () => {
+            test('@includes on interfaces', () => {
+                query = `
+                    query {
+                        hero(episode: EMPIRE) {
+                            ...@include(if: true) {
+                                name
+                                friends(first: 3) {
+                                    name
+                                }
+                            }
+                            ... on Human {
+                                homePlanet
+                            }
+                        }
+                    }`;
+                mockCharacterFriendsFunction.mockReturnValueOnce(3);
+                // Query 1 + 1 hero + max(...Character 3, ...Human 0) = 5
+                expect(queryParser.processQuery(parse(query))).toBe(5);
+
+                query = `
+                    query {
+                        hero(episode: EMPIRE) {
+                            ...@include(if: false) {
```

Comment:
I would expect this function wouldn't even be called. Could be worth adding an assertion for this to avoid unnecessary work.